### PR TITLE
Fine tune pelias api 

### PR DESCRIPTION
**Pull request by vesameskanen**
- Confidence scoring of non-parsed input strings improved. Simple searches like 'Apteekki' now score better against composite names such as 'Pakilan apteekki'.

- Some new libpostal parsing errors fixed:
1. Don't accept city name parsed as a postal code (i.e. postal code validation). Example: helsinki, helsinki 
2. Sometimes parsed name  is incorrectly used as a neighbourhood. Example: siwa, helsinki. Fixed by erasing multiple identical search parameters.

### CODE REVIEW

**Comment by hannesj** [comment]
We should probably open at least two issues in https://github.com/openvenues/libpostal about these things

File: sanitizer/_text_addressit.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ function assignValidLibpostalParsing(parsedText, fromLibpostal, text) {
     if(check.assigned(fromLibpostal.neighbourhood)) {
       var nbrh = restoreParsed(fromLibpostal.neighbourhood, text);
 
-      if(nbrh) {
+      if(nbrh && parsedText.name !== nbrh) { // don't add same string to both name and admin parts
```
Comment (vesameskanen) [inline sanitizer/_text_addressit.js]: Yes, it does - our data includes places where name===malminkartano.

**Comment by vesameskanen** [comment]
It seems that libpostal fails in parsing in quite a random manner. Any address part may get incorrectly assigned to some other address part. Sometimes name->street, city->street, street->city, name->postalcode, etc . The form of the address seems to play only a little role; names consisting of several words are naturally more difficult. I guess it is the NLP approach that causes this kind of behavior (not enough training with finnish data).

Parsing failure rate is quite high, maybe 10 % or so. Our implementation works decently because it is possible to detect libpostal parsing failures.  If parsing looks suspicious, geocoder uses the simple addressit parsing.

File: sanitizer/_text_addressit.js [hunk 2]
```diff
@@ -13,10 +13,17 @@ var api = require('pelias-config').generate().api;
...
 var filteredRegions;
 var cleanRegions;
+var postalCodeValidator = function(code) { return true; }; // default = accept everything
 
 if (api && api.localization) {
   filteredRegions = api.localization.filteredRegions;
   cleanRegions = api.localization.cleanRegions;
+  if(api.localization.postalCodeValidator) {
+    postalCodeValidator = function(code) {
+      var regexp = new RegExp(api.localization.postalCodeValidator);
```
Comment (vesameskanen) [inline sanitizer/_text_addressit.js]: Indeed! Fixed now.

**Pull request merged**